[pytree] Allow register_pytree_node to take in 5 inputs

angelayi opened the pull request:
Summary: This is currently breaking internal old torch.packages from when _register_pytree_node took 5 inputs.

Test Plan: CI

Differential Revision: D48834742

zou3519 commented on torch/utils/_pytree.py:
```diff
@@ -72,7 +72,6 @@ def _register_pytree_node(
     typ: Any,
     flatten_fn: FlattenFunc,
     unflatten_fn: UnflattenFunc,
-    *,
```
Are we ever able to undo this change?

angelayi replied:
Yes, but whenever torch packages become outdated internally...but I'm not sure when that is. We will remove the call that is passing in positional args in D48784113, so future migration will hopefully be easier...

zou3519 reviewed (approved):
okey doke

angelayi commented:
@zou3519 should this change be in the 2.1 release?

zou3519 commented:
The ideal state is if we can undo this change. If we cannot undo this change, then this should go into the branch cut so that the API is consistent

angelayi committed:
[pytree] Allow register_pytree_node to take in 5 inputs (#108256)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/108256

This is currently breaking old torch.packages from when _register_pytree_node took 5 inputs.

Test Plan: CI

Reviewed By: zou3519, PaulZhang12, zhxchen17

Differential Revision: D48834742

fbshipit-source-id: 16286e105197392ee95ea2b06811206738a74235
